process/block/sovereignChainBlock_test.go

```diff
@@ -343,4 +400,283 @@ func TestSovereignChainBlockProcessor_createAndSetOutGoingMiniBlock(t *testing.T
+		busyIdleCalled := make([]string, 0)
+		mockProcessHandler.SetIdleCalled = func() {
+			busyIdleCalled = append(busyIdleCalled, idleIdentifier)
+		}
+		mockProcessHandler.SetBusyCalled = func(reason string) {
+			busyIdleCalled = append(busyIdleCalled, busyIdentifier)
+		}
+		expectedBusyIdleSequencePerCall := []string{busyIdentifier, idleIdentifier}
+
+		sovArgs := createArgsSovereignChainBlockProcessor(arguments)
+		scbp, err := blproc.NewSovereignChainBlockProcessor(sovArgs)
+		require.Nil(t, err)
+
+		doesHaveTime := func() bool {
+			return true
+		}
+		sovHeader := &block.SovereignChainHeader{
+			Header: &block.Header{
+				Nonce:         1,
+				PubKeysBitmap: []byte("0100101"),
+				PrevHash:      []byte(""),
+				PrevRandSeed:  []byte("rand seed"),
+				Signature:     []byte("signature"),
+				RootHash:      []byte("roothash"),
+			},
+		}
+
+		hdr, body, err := scbp.CreateBlock(sovHeader, doesHaveTime)
+		require.True(t, check.IfNil(body))
+		require.True(t, check.IfNil(hdr))
+		require.Equal(t, process.ErrAccountStateDirty, err)
+		require.Equal(t, expectedBusyIdleSequencePerCall, busyIdleCalled)
+	})
+	t.Run("create block started should error", func(t *testing.T) {
+		expectedErr := fmt.Errorf("createBlockStarted error")
+
+		coreComponents, dataComponents, bootstrapComponents, statusComponents := createComponentHolderMocks()
+		arguments := createSovereignMockArguments(coreComponents, dataComponents, bootstrapComponents, statusComponents)
+		arguments.TxCoordinator = &testscommon.TransactionCoordinatorMock{
+			AddIntermediateTransactionsCalled: func(_ map[block.Type][]data.TransactionHandler, _ []byte) error {
+				return expectedErr
+			},
+		}
+		sovArgs := createArgsSovereignChainBlockProcessor(arguments)
+		processHandler := arguments.CoreComponents.ProcessStatusHandler()
+		mockProcessHandler := processHandler.(*testscommon.ProcessStatusHandlerStub)
+		busyIdleCalled := make([]string, 0)
+		mockProcessHandler.SetIdleCalled = func() {
+			busyIdleCalled = append(busyIdleCalled, idleIdentifier)
+		}
+		mockProcessHandler.SetBusyCalled = func(reason string) {
+			busyIdleCalled = append(busyIdleCalled, busyIdentifier)
+		}
+		expectedBusyIdleSequencePerCall := []string{busyIdentifier, idleIdentifier}
+
+		scbp, err := blproc.NewSovereignChainBlockProcessor(sovArgs)
+		require.Nil(t, err)
+
+		doesHaveTime := func() bool {
+			return true
+		}
+
+		expectedSovHeader := &block.SovereignChainHeader{
+			Header: &block.Header{
+				Nonce: 37,
+				Round: 38,
+				Epoch: 1,
+			},
+		}
+
+		hdr, bodyHandler, err := scbp.CreateBlock(expectedSovHeader, doesHaveTime)
+		require.True(t, check.IfNil(bodyHandler))
+		require.True(t, check.IfNil(hdr))
+		require.Equal(t, expectedErr, err)
+		require.Equal(t, expectedBusyIdleSequencePerCall, busyIdleCalled)
+	})
+	t.Run("should work with sovereign header and epoch start rewriting the epoch value", func(t *testing.T) {
```

Comment:
Please rename this func name, we do not rewrite, I think you mean we change epoch